cmd/segment-reaper/detect.go

```diff
@@ -25,9 +34,126 @@ var (
 		DatabaseURL string `help:"the database connection string to use" default:"postgres://"`
 		From        string `help:"begin of date range for detecting zombie segments" default:""`
 		To          string `help:"end of date range for detecting zombie segments" default:""`
+		File        string `help:"location of file with report" default:"detect_result.out"`
 	}
 )
 
+// Cluster key for objects map
+type Cluster struct {
+	projectID string
+	bucket    string
+}
+
+// Object represents object with segments
+type Object struct {
+	// TODO verify if we have more than 64 segments for object in network
+	segments uint64
+
+	expectedNumberOfSegments byte
+
+	hasLastSegment bool
+	// if skip is true segments from object should be removed from memory when last segment is found
+	// or iteration is finished, mark it as true if one of the segments from this object is newer
+	// then specified threshold
+	skip bool
+}
+
+// ObjectsMap map that keeps objects representation
+type ObjectsMap map[Cluster]map[string]*Object
+
+// Observer metainfo.Loop observer for zombie reaper
+type Observer struct {
+	db      metainfo.PointerDB
+	objects ObjectsMap
+	writer  *csv.Writer
+
+	lastProjectID string
+
+	inlineSegments     int
+	lastInlineSegments int
+	remoteSegments     int
+}
+
+// RemoteSegment takes a remote segment found in metainfo and creates a reservoir for it if it doesn't exist already
```

Comment:
This comment looks like copy-pasted and not relevant for the usage here. Same for the other method comments of the observer.

Reply:
yes, fixing